test/TestProjects/CadlFirstTest/Generated/Demo2Client.cs

```diff
@@ -375,7 +375,7 @@ internal HttpMessage CreateHelloDemo2Request(RequestContext context)
         private static RequestContext DefaultRequestContext = new RequestContext();
         internal static RequestContext FromCancellationToken(CancellationToken cancellationToken = default)
         {
-            if (cancellationToken == CancellationToken.None)
+            if (!cancellationToken.CanBeCanceled)
```

Comment:
@annelo-msft IMO, it is better to use a property rather than compare struct to a default value. But I can revert it, up to you.

Reply:
I believe we had this discussion and decided CanBeCancelled has [too many nuances](https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken.canbecanceled?view=net-6.0#remarks) and what we want here is to compare it to CancellationToken.None instead of checking CanBeCancelled.  @JoshLove-msft, @tg-msft, do you remember this conversation, and can you help me confirm?

Reply:
For example, `CancellationTokenSource` uses `CanBeCancelled`: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs#L737

Reply:
I thought the recommendation was to use CanBeCancelled rather than comparing to None.